src/box/box.cc

```diff
@@ -1375,10 +1375,22 @@ box_process1(struct request *request, box_tuple_t **result)
 	struct space *space = space_cache_find(request->space_id);
 	if (space == NULL)
 		return -1;
+	struct engine *engine = (struct engine *)space->engine;
```

Comment:
space->engine is already of type (struct engine *), so cast is redundant. Also, I'd move this assignment closer to first usage of var engine, i.e. right after space_is_temporary() check.